ballerina/tests/mock_service.bal

```
@@ -252,56 +157,48 @@ service on new http:Listener(9090) {
-                        "additionalProp1": "string",
-                        "additionalProp2": "string",
-                        "additionalProp3": "string"
+                    createdAt: "2025-02-17T04:04:47.711Z",
+                    archived: true,
+                    archivedAt: "2025-02-17T04:04:47.711Z",
+                    'new: true,
+                    propertiesWithHistory: {},
+                    id: "string",
+                    properties: {
+                        "hs_timestamp": "2025-02-17T01:32:44.872Z",
+                        "hs_call_title": "Support call",
+                        "hubspot_owner_id": hsOwnerId,
+                        "hs_call_body": "Resolved issue",
+                        "hs_call_duration": "3800",
+                        "hs_call_from_number": "(857) 829 5489",
+                        "hs_call_to_number": "(509) 999 9999",
+                        "hs_call_recording_url": "example.com/recordings/abc",
+                        "hs_call_status": "COMPLETED"
                     },
-                    "updatedAt": "2025-02-17T04:04:47.711Z"
+                    updatedAt: "2025-02-17T04:04:47.711Z"
                 }
             ]
         };
         return response;
     }
 
-    resource function get [string callId](http:Request req) returns SimplePublicObjectWithAssociations {
+    resource function get [string callId](http:Request req) returns SimplePublicObjectWithAssociations|http:Response|error {
+        string id = "";
+        lock {
+            // find the id from the callIds array  
+            foreach string i in callIds {
+                if i == callId {
+                    id = i;
+                    break;
+                }
+            }
+        }
+        if id == "" {
+            http:Response response = new;
+            response.statusCode = http:STATUS_NOT_FOUND;
+            response.setPayload({"error": "Call ID not found"});
+            return response;
```

we can just return an error here.

```suggestion
            return error("Call ID not found");
```

---

Sure, but when I used `return error("Call ID not found");`  instead of http response object following message appears in the console. That why I used a http response object to return errors.

`time=2025-02-22T17:34:09.653+05:30 level=ERROR module=ballerina/http message="unhandled error returned from the service" error={"causes":[],"message":"Call ID not found","detail":{},"stackTrace":[{"callableName":"$get$^","moduleName":"ballerinax.hubspot.crm.engagements.calls$test.1.$anonType$_26","fileName":"tests/mock_service.bal","lineNumber":196}]} path="/invalid_call_id?archived=false" method="GET"`